overlord/ifacestate/handlers.go

```diff
@@ -744,6 +744,64 @@ func (m *InterfaceManager) doAutoConnect(task *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+// doDisconnectInterfaces creates tasks for disconnecting an interface and running its interface hooks.
+func (m *InterfaceManager) doDisconnectInterfaces(task *state.Task, _ *tomb.Tomb) error {
```

Comment:
unless I'm confused, this has no unit tests in the package atm

Reply:
Tests added.